File: IntegracaoJavaScript/split.js
const pagarme = require('pagarme')

pagarme.client.connect({ api_key: 'ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg' })
.then(client => client.transactions.create({
    "amount": 1000,
    "card_id": "card_ck394f18e01flvh6d3sdtoohj",
    "postback_url": "http://requestb.in/pkt7pgpk",
    "split_rules":[
    	{
    	"liable": "true",
    	"charge_processing_fee": "true",
    	"percentage": "90",
    	"charge_remainder_fee": "true",
    	"recipient_id":"re_ck0zkjmuj00gln86dajtwrfws"
    	},
    	{
    	"liable": "false",
    	"charge_processing_fee": "false",
    	"percentage": "10",
    	"charge_remainder_fee": "false",
    	"recipient_id":"re_ck0zjmw1s007ier6eysyd3agj"
    	}
    	],
    "customer": {
      "external_id": "#3311",
      "name": "Morpheus Fishburne",
      "type": "individual",
      "country": "br",
      "email": "mopheus@nabucodonozor.com",
      "documents": [
        {
          "type": "cpf",
          "number": "30621143049"
        }
      ],
      "phone_numbers": ["+5511999998888", "+5511888889999"],
      "birthday": "1965-01-01"
    },
    "billing": {
      "name": "Trinity Moss",
      "address": {
        "country": "br",
        "state": "sp",
        "city": "Cotia",
        "neighborhood": "Rio Cotia",
        "street": "Rua Matrix",
        "street_number": "9999",
        "zipcode": "06714360"
      }
    },
    "shipping": {
      "name": "Neo Reeves",
      "fee": 1000,
      "delivery_date": "2000-12-21",
      "expedited": true,
      "address": {
        "country": "br",
        "state": "sp",
        "city": "Cotia",
        "neighborhood": "Rio Cotia",
        "street": "Rua Matrix",
        "street_number": "9999",
        "zipcode": "06714360"
      }
    },
    "items": [
      {
        "id": "r123",
        "title": "Red pill",
        "unit_price": 10000,
        "quantity": 1,
        "tangible": true
      },
      {
        "id": "b123",
        "title": "Blue pill",
        "unit_price": 10000,
        "quantity": 1,
        "tangible": true
      }
    ]
}))
.then(transaction => { console.log(transaction.id)})

File: IntegracaoCSharp/CSharp-test2/Split.cs
using System;
using PagarMe;

namespace CSharptest2
{
    public class Split
    {
        public String TSplit()
        {
            PagarMeService.DefaultApiKey = "ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg";
            PagarMeService.DefaultEncryptionKey = "ek_test_SF2MPmKaWRLywNuetObE0vkv0nnkel";

            Transaction transaction = new Transaction();

            transaction.Amount = 2000;
            var Card = new Card
            {
                Number = "4018720572598048",
                HolderName = "Aardvark Silva",
                ExpirationDate = "1125",
                Cvv = "123"
            };
            Card.Save();
            transaction.Card = Card;
            transaction.SplitRules = new SplitRule[]{
                new SplitRule
                {
                    Percentage = 80,
                    RecipientId = "re_ck0zkjmuj00gln86dajtwrfws",
                    ChargeProcessingFee = true,
                    Liable = true
                },
                new SplitRule
                {
                    Percentage = 20,
                    RecipientId = "re_ck0zjmw1s007ier6eysyd3agj",
                    ChargeProcessingFee = false,
                    Liable = false
                }
            };
            transaction.Customer = new Customer
            {
                ExternalId = "#3311",
                Name = "Rick",
                Type = CustomerType.Individual,
                Country = "br",
                Email = "rick@morty.com",
                Documents = new[]
            {
                new Document{
                Type = DocumentType.Cpf,
                Number = "30621143049"
                },
                new Document{
                Type = DocumentType.Cnpj,
                Number = "83134932000154"
                }
            },
                PhoneNumbers = new string[]
                {
                "+5511982738291",
                "+5511829378291"
                },
                Birthday = new DateTime(1991, 12, 12).ToString("yyyy-MM-dd")
            };

            transaction.Billing = new Billing
            {
                Name = "Morty",
                Address = new Address()
                {
                    Country = "br",
                    State = "sp",
                    City = "Cotia",
                    Neighborhood = "Rio Cotia",
                    Street = "Rua Matrix",
                    StreetNumber = "213",
                    Zipcode = "04250000"
                }
            };

            var Today = DateTime.Now;

            transaction.Shipping = new Shipping
            {
                Name = "Rick",
                Fee = 100,
                DeliveryDate = Today.AddDays(4).ToString("yyyy-MM-dd"),
                Expedited = false,
                Address = new Address()
                {
                    Country = "br",
                    State = "sp",
                    City = "Cotia",
                    Neighborhood = "Rio Cotia",
                    Street = "Rua Matrix",
                    StreetNumber = "213",
                    Zipcode = "04250000"
                }
            };

            transaction.Item = new[]
            {
            new Item()
                {
                    Id = "1",
                    Title = "Little Car",
                    Quantity = 1,
                    Tangible = true,
                    UnitPrice = 1000
                },
            new Item()
                {
                    Id = "2",
                    Title = "Baby Crib",
                    Quantity = 1,
                    Tangible = true,
                    UnitPrice = 1000
                }
            };

            transaction.Save();
            return Convert.ToString(transaction.Id);
        }
    }
}

